### ISSUE TITLE
Mage clearcasting not procing correctly with blizzard ONLY, all other skills work.

### ISSUE THREAD

**Issue by Josephina1234** [chromiecraft/chromiecraft#2163]
### What client do you play on?

enGB

### Faction

- [ ] Alliance
- [X] Horde

### Content Phase:

- [ ] Generic
- [ ] 1-19
- [ ] 20-29
- [ ] 30-39
- [ ] 40-49
- [X] 50-59

### Current Behaviour

Clearcasting has chance to proc ONLY on first hit of a channeling spell like blizzard which is wrong. 

### Expected Blizzlike Behaviour

Clearcasting should be able to proc on every tic of blizzard's full channel ( can proc multiple times during single cast) like it worked in all previous brackets but just recently it got changed. 

### Source
Proof that blizzard should have 10% chance to proc on every blizzard wave 
https://wowwiki-archive.fandom.com/wiki/Arcane_Concentration

U can read "Further, Blizzards often proc clearcasting when allowed to run to their full duration, due to the large number of times Blizzard strikes and thus has a chance at proc'ing, which can sometimes mean that you can chain two, three, or even four Blizzards for no mana cost that all (combined with Arcane Potency) have a +30% crit chance. It's a fatal barrage.

Unlike "on spell hit" effects, Arcane Concentration cannot proc off each hit of an Arcane Missiles' bolt, but only off the initial hit of AM itself"

So every hit of blizzard can proc it, but not arcane missiles. 

With history option u can look that is how it worked in original wotlk.

### Steps to reproduce the problem

1. Have Mage level 59
2. Spec like this https://wotlkdb.com/?talent#of0V0zZZAccIs0u0czTb
3. Attack 20 mobs with blizzard
4. If clearcasting dont proc on first tic of damage, it will not proc at all during full cast. 

### Extra Notes

Clearcasting talent: https://wotlkdb.com/?talent#of0V0zZZAccIs0u0czTb
Blizzard spell: https://wotlkdb.com/?spell=42940

### AC rev. hash/commit

https://github.com/chromiecraft/azerothcore-wotlk/commit/e9c71c8ebefc067e3119a2a1a63b74c3239cc55c

### Operating system

Ubuntu 20.04

### Modules

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-chromie-xp](https://github.com/azerothcore/mod-chromie-xp)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [lua-carbon-copy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-custom-worldboss](https://github.com/55Honey/Acore_CustomWorldboss)
- [lua-level-up-reward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-recruit-a-friend](https://github.com/55Honey/Acore_RecruitAFriend)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-temp-announcements](https://github.com/55Honey/Acore_TempAnnouncements)
- [lua-zonecheck](https://github.com/55Honey/acore_Zonecheck)

### Customizations

None

### Server

ChromieCraft

**Comment by bonfax4** [comment]
"Arcane Concentration: The interaction of this talent with Arcane Missiles has been restructured. Instead of an Arcane Missiles cast triggering the talent, each missile has an independent (but reduced) chance to trigger this talent. In addition, the talent will no longer sometimes be triggered from misses by Arcane Missiles, Arcane Blast, and Blizzard." I find it weird that blizzard is mentioned here on this specific change in 3.3 like the chance is not being based on cast but on something else entirely. Even the way missiles currently work is incorrect now since it has a chance on each missile and not on cast. It might be @2% per missile hit instead of 10% on cast.

**Comment by bonfax4** [comment]
Arcvane explosion might need a look to according to this post https://www.mmo-champion.com/threads/682143-How-and-When-to-AoE-as-Arcane-Mage (Date would put this discussion around patch 3.3)
"This. On large groups of mobs, due to the way Clearcasting operates, Clearcasting will be up almost every time you hit your Arcane Explosion key, meaning you can spam it whilst using virtually no mana. On top of this every Clearcasting poc also procs your 30% increased crit chance, meaning your Arcane Explosion crit chance will likely be around 70% or more."

It's not that crazy to believe it works like this since fire mage rn also has infinite mana with ignite during aoe pulls but i'll try to research a bit more. Would be easier if worldoflogs was still up :/

**Comment by bonfax4** [comment]
**For blizzard spell (retail video):** https://youtu.be/Zk8F2lY1aHY?t=124 ( slow down teh video to 0.5 ) It's def not a per number of targets based on this video otherwise it would be up 100% of the time but it does seem to proc mid cast so it can be related to a  "tick" of the cast. Retail video btw.

**For Arcane Explosion:** https://youtu.be/FIDEjgQPjGE?t=88 Proccing very very often, consistent with the post linked in my comment above.

**Comment by Gultask** [comment]
>Further, Blizzards often proc clearcasting when allowed to run to their full duration, due to the large number of times Blizzard strikes and thus has a chance at proc'ing, which can sometimes mean that you can chain two, three, or even four Blizzards for no mana cost that all (combined with Arcane Potency) have a +30% crit chance. It's a fatal barrage. ([source](https://wowpedia.fandom.com/wiki/Arcane_Concentration?oldid=2109239))

**Comment by bonfax4** [comment]
Had comments on this but deleted them since I noticed the retail videos I linked where clearcasting seemed to proc seeminggly mid cast of a blizzard might have happened due to a molten armor tick of damage. So I actually do believe now that it procs on either the 1st wave hit ( not per target ) or the begining of a cast. Weirdly enough tho it does proc per target on arcane explosion. Wowwiki has proved to be unreliable to me tbh sometimes info is not updated properly. There might have been some wotlk beta where this was possible but didn't follow through.

**Comment by Gultask** [comment]
Yes I saw another video on Onyxia too where AE was proccing every single time, and a lot of people mentioned that in forums too.
And that video I posted showed clearcasting proccing near the end of the Blizzard channeling so I think it's safe.

**Comment by Josephina1234** [comment]
Will this ever be fixed, its been almost a year since i reported it. Worse is you had it working perfectly fine for a long time and than someone started acting smart and using TBC as evidence of how skills work in WOTLK despite clear evidence that in wotlk its different.

**Comment by Salehyn** [comment]
My latest tests on CC PTR shows some change was applied here. It procs on ticks now, seemingly always the first set, and the chance seems to scale with target count. It never seems to second tick and onwards.

**Comment by Salehyn** [comment]
Testing on CC PTR currently shows that while only the first tick is capable of proccing the Clearcast, the proc chance scales with the target count hit (each target has its own 10% roll, so fairly good odds on 4+ targets). As I understand this is not possible to fix at the moment (based on the comments on the AC thread) which is why the spell was left as-is for now. Allowing blizzard to proc on each tick would give a guaranteed proc on each cast if the target issue isn't addressed at the same time.
